python/lsst/ts/electrometer/controller.py

```diff
@@ -333,47 +360,115 @@ async def start_scan_dt(self, scan_duration):
+        primary_hdu.header["SCANTIME"] = (self.scan_duration, "Duration of scan [s]")
+        primary_hdu.header["SAMPTIME"] = (
+            self.integration_time,
+            "Duration of each sample [s]",
+        )
+        primary_hdu.header["FILTMED"] = (
+            self.median_filter_active,
+            "Median Filter Active",
+        )
+        primary_hdu.header["FILTAVG"] = (
+            self.avg_filter_active,
+            "Average Filter Active",
+        )
+        primary_hdu.header["IMGTYPE"] = (
+            self.mode,
+            "Options are charge, voltage, current",
+        )
+        primary_hdu.header["SENSBRND"] = (self.sensor_brand, "Sensor brand")
+        primary_hdu.header["SENSMODL"] = (self.sensor_model, "Sensor model")
+        primary_hdu.header["SERIAL"] = (self.sensor_serial, "Sensor serial number")
+        primary_hdu.header["TEMP"] = (
+            self.temperature,
+            "Measurement from probe if attached and declared (Celcius)",
+        )
+        primary_hdu.header["VSOURCE"] = (
+            self.vsource,
+            "Voltage input if active and attached",
+        )
+        return primary_hdu
+
+    async def write_fits_file(self, signal, times, temperature, unit, voltage):
         """Write fits file of the intensity, time, and temperature values.
 
         Parameters
         ----------
-        intensity : `list`
-            The intensity values
+        signal : `list`
+            The signal values.
         times : `list`
-            The time of the intensity value
+            The time of the signal data taken.
         temperature : `list`
-            The temperature of the intensity value
+            The temperature of the signal data. (constant)
         unit : `list`
-            The unit of the intensity value.
+            The unit of the signal data. (constant)
+        voltage : `list`
+            The voltage source input. (constant)
         """
-        data = np.array([times, intensity])
-        hdu = fits.PrimaryHDU(data)
-        hdr = hdu.header
-        hdr["CLMN1"] = ("Time", "Time in seconds")
-        hdr["CLMN2"] = "Intensity"
+        if self.temperature_attached:
+            self.temperature = temperature[0]
+        if self.voltage_status:
+            self.vsource = voltage[0]
+        primary_hdu = self.make_primary_header()
+        data_columns = [signal, times]
+        data_names = ["Elapsed Time", "Signal"]
+        data_metadata = {"name": "Single Electrometer scan readout"}
+
+        data_table = table.QTable(
+            data=data_columns, names=data_names, meta=data_metadata
+        )
+        table_hdu = fits.table_to_hdu(data_table)
+        hdul = fits.HDUList([primary_hdu, table_hdu])
         filename = f"{self.manual_start_time}_{self.manual_end_time}.fits"
         try:
-            hdu.writeto(f"{self.file_output_dir}/{filename}")
+            hdul.writeto(f"{self.file_output_dir}/{filename}")
```

Comment:
This will fail if the `self.file_output_dir` doesn't exist. Please add a check that verifies that it exists and that creates the dir if it doesn't.

Reply:
You added the check and log an error. Why not add code that created the directory as well?